server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -180,10 +180,15 @@ protected void run(Object batchingKey, List<? extends BatchedTask> tasks, String
 
```

Comment:
It seems like we can not remove the listener as it is wrapped with `safe` during the task creation at https://github.com/elastic/elasticsearch/blob/b482576b152f6fd2a2c41bb5312bdc3b2fb256f4/server/src/main/java/org/elasticsearch/cluster/service/MasterService.java#L925

Reply:
Yes I think that's right for now at least. This prompted me to realise that we're missing something similar in a few other places: #83019, #83021, #83035. Fortunately I don't think anything relies on the context being correct in those places (... yet).